### PR TITLE
ROX-9462 Collect baselines

### PR DESCRIPTION
**Pull request by erthalion**
Introduce a script to manage performance baselines numbers. The baseline
is essentially a json file, containing previous benchmark results. Its
size is limited, the exact threshold has to be figured out depending on
the actual variance of data.

The script supposed to collect baseline, and provide way to compare
current benchmark data with it. Due to unavoidable noise try to be fancy
and employ t-test for mean values to figure out if the difference we've
got is significant, or just a noise.

This change also introduces a slight change in `all-perf.json` format,
before it was a list of json documents concatenated together, like:

    {"aaa": 1, "bbb": 2, ...} {"aaa": 3, "bbb": 4, ...}

This was fine for jq, but it's not a valid json. Not it will be:

    [{"aaa": 1, "bbb": 2, ...}, {"aaa": 3, "bbb": 4, ...}]

jq apparently didn't notice a thing.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

## Testing Performed

Only local testing.

### CODE REVIEW

File: .circleci/test-scripts/baseline/main.py
Comment (Molter73) [inline .circleci/test-scripts/baseline/main.py:182]: Couldn't these two list comprehensions be combined if we moved the if condition of the first one to the last one?

File: integration-tests/integration_test.go
Comment (Molter73) [inline integration-tests/integration_test.go:435]: [nit] What happened to the spacing here? Seems like some auto-formatting might be playing some nasty tricks

**Review by Molter73 — APPROVED** [review]
Just some minor comments that wouldn't be blocker IMO. LGTM!

File: integration-tests/integration_test.go
Comment (erthalion) [reply integration-tests/integration_test.go:435]: Looks like an interesting [quirk](https://github.com/golang/go/issues/1206) of gofmt. When I run `gofmt` directly over this file it doesn't report any difference, and it seems like it does remove spaces sometimes "to express binding strength".

File: .circleci/config.yml
Comment (erthalion) [reply .circleci/config.yml:1084]: It's not really a full re-write, there are going to be some follow up steps for sure. But I don't have anything against moving it into a script :)

**Comment by erthalion** [comment]
There is only one thing that bothers me at the moment -- since the baseline file is simply an array of benchmark results, there is no explicit connections between `baseline_benchmark` and corresponding `collector_benchmark`. It works now via sorting results  by timestamps and correlating things in this way, which is a fragile approach. I believe this could be improved as a follow-up, when we get more data to see if it's a problem or not.

File: .circleci/test-scripts/baseline/compare-and-update-baseline.sh
Comment (Molter73) [inline .circleci/test-scripts/baseline/compare-and-update-baseline.sh:2]: Might be worth it to add `-u` in order for the script to fail explicitly if some env variable isn't defined.
```suggestion
set -euo pipefail
```

File: .circleci/test-scripts/baseline/compare-and-update-baseline.sh
Comment (Molter73) [inline .circleci/test-scripts/baseline/compare-and-update-baseline.sh:20]: Why was this left as a comment? Is it replacing line 13 before merging the PR?

**Review by Molter73 — APPROVED** [review]
Re-approving after the latest changes. Added 2 comments but they are super-nit things that shouldn't prevent this from being merged.

File: .circleci/test-scripts/baseline/compare-and-update-baseline.sh
Comment (erthalion) [reply .circleci/test-scripts/baseline/compare-and-update-baseline.sh:20]: Yep, that's the goal.

**Review by Stringy — APPROVED** [review]
re-approving, looks good!

**Pull request merged**
